thunder/tests/test_jit_general.py

```diff
@@ -813,3 +813,45 @@ def foo(a, scalar):
     assert_close(actual, expected)
     assert thunder.cache_misses(jfoo) == 1
     assert thunder.cache_hits(jfoo) == 1
+
+
+def test_post_optimization_transform():
+    def foo(a, b, c):
+        return a * a + b * c
+
+    class MyTransform(PostOptimizationTransform):
+        def __call__(self, fwd_trc, bwd_trc, executors_list=None):
+            # Transform that adds a comment before any `add` BoundSymbol.
+            def _comment_add_op_transform(trace):
+                commented_trace = thunder.core.trace.from_trace(trace)
+
+                bsyms = []
+                for bsym in trace.bound_symbols:
+                    if bsym.sym.name == "add":
+                        op_name = bsym.sym.name
+                        comment_bsym = prims.comment.bind(f"Executing {op_name}", output=None)
+                        bsyms.append(comment_bsym)
+
+                    bsyms.append(bsym)
+
+                commented_trace.bound_symbols = bsyms
+                return commented_trace
+
+            return _comment_add_op_transform(fwd_trc), (
+                _comment_add_op_transform(bwd_trc) if bwd_trc is not None else None
+            )
```

Comment:
I think the transform should take just one trace and "if backward is not None" should be in infra code not in transform code.

Reply:
Makes sense, also linking the relevant [comment](https://github.com/Lightning-AI/lightning-thunder/issues/533#issuecomment-2152894973) from the issue here